### PR TITLE
Clarify futher  the order for a rolling upgrade

### PR DESCRIPTION
**Pull request by DaveCTurner**
Expands the "master-ineligible then master-eligible" sentence into a list and
specifies that within these subsets the order doesn't matter.

### CODE REVIEW

**Review by eedugon — APPROVED** [review]
LGTM

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 1]
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
 not supported, as shards cannot be replicated from upgraded nodes to nodes
 running the older version.
 
-It is best to upgrade the master-eligible nodes in your cluster after all of
-the other nodes. Once you have started to upgrade the master-eligible nodes
-they may form a cluster that nodes of older versions cannot join. If you
-upgrade the master-eligible nodes last then all the other nodes will not be
-running an older version and so they will be able to join the cluster.
+It is strongly recommended to upgrade the nodes in your cluster in the
+following order:
```
Comment (jrodewig) [inline docs/reference/upgrade/rolling_upgrade.asciidoc]: I'd use direct voice here. It makes the sentence a bit stronger.

> We strongly recommend you upgrade your cluster's nodes in the following order:

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 2]
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
...
+
+. Nodes that are not master-eligible. You can retrieve a list of these nodes
```
Comment (jrodewig) [inline docs/reference/upgrade/rolling_upgrade.asciidoc]: I would link to master-eligible here.

```suggestion
. Nodes that are not <<master-node,master-eligible>>. You can retrieve a list of these nodes
```

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 3]
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
...
+with `GET /_nodes/master:false`, or by finding all the nodes which are
```
Comment (jrodewig) [inline docs/reference/upgrade/rolling_upgrade.asciidoc]: This comma is unneeded.
```suggestion
with `GET /_nodes/master:false` or by finding all the nodes
```

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 4]
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
...
+configured with `node.master: false`. You may upgrade this subset of your nodes
```
Comment (jrodewig) [inline docs/reference/upgrade/rolling_upgrade.asciidoc]: I don't think "your" is needed here.

> configured with `node.master: false`. You may upgrade this subset of nodes

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 5]
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
...
+in any order.
+
+. Nodes that are master-eligible, which is all the remaining nodes. You can
```
Comment (jrodewig) [inline docs/reference/upgrade/rolling_upgrade.asciidoc]: I'd reword this slightly:

> Master-eligible nodes, which are all remaining nodes in the cluster.

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 6]
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
...
+retrieve a list of these nodes with `GET /_nodes/master:true`. You may upgrade
+this subset of your nodes in any order.
```
Comment (jrodewig) [inline docs/reference/upgrade/rolling_upgrade.asciidoc]: "your" is unneeded.

> this subset of nodes in any order.

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 7]
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
...
+
+The reason for this recommendation is that newer nodes can always join a
+cluster with an older master, but older nodes cannot always join a cluster with
+a newer master. By upgrading the master-eligible nodes last you ensure that all
+the other nodes are running the newer version first and so they will be able to
+join the cluster whether the master-eligible nodes have been upgraded or not.
```
Comment (jrodewig) [inline docs/reference/upgrade/rolling_upgrade.asciidoc]: I believe this sentence is stronger without "The reason for ..." and some rewording:

> Newer nodes can always join a cluster with an older master, but older nodes cannot always join a cluster with
a newer master. This upgrade approach ensure all other nodes are first running a newer version. This means they can join the cluster, even if the cluster's master-eligible nodes have not been upgraded.

**Review by jrodewig — APPROVED** [review]
This looks pretty good. I left some suggestions for rewording which you can take or leave as you see fit.

I think all the crucial info is here. Thanks again for raising this.

**Review by inqueue — APPROVED** [review]
LGTM

**Comment by crisdarocha** [comment]
LGTM, specially with the comments from @jrodewig .

**Review by crisdarocha — APPROVED** [review]
LGTM specially with @jrodewig changes

**Review by richiejarvis — DISMISSED** [review]
Hi All,

I do not think that having the phrase "You may upgrade this subset of your nodes in any order" after each one is clear enough.  How about putting this more general text above?

"The order in which you upgrade each node in each subset is not important."

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 8]
```diff
@@ -7,11 +7,20 @@ a time so upgrading does not interrupt service. Running multiple versions of
 not supported, as shards cannot be replicated from upgraded nodes to nodes
 running the older version.
 
-It is best to upgrade the master-eligible nodes in your cluster after all of
-the other nodes. Once you have started to upgrade the master-eligible nodes
-they may form a cluster that nodes of older versions cannot join. If you
-upgrade the master-eligible nodes last then all the other nodes will not be
-running an older version and so they will be able to join the cluster.
+It is strongly recommended to upgrade the nodes in your cluster by grouping the nodes into two groups, as shown below.  Nodes within each group may be upgraded in any order, as long as the last group is the master-eligable group:
```
Comment (jasontedor) [inline docs/reference/upgrade/rolling_upgrade.asciidoc]: Nit: please wrap at 80 columns.

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 9]
```diff
@@ -7,11 +7,20 @@ a time so upgrading does not interrupt service. Running multiple versions of
...
+
+. Nodes that are not master-eligible. You can retrieve a list of these nodes
+with `GET /_nodes/master:false`, or by finding all the nodes which are
+configured with `node.master: false`.
+
+. Nodes that are master-eligible, which is all the remaining nodes. You can
+retrieve a list of these nodes with `GET /_nodes/master:true`.
+
+By upgrading in this order it ensures that unforeseen circumstances do not break the upgrade.  
+Newer nodes can always join a cluster with an older master, but older nodes cannot always join a cluster with
```
Comment (jasontedor) [inline docs/reference/upgrade/rolling_upgrade.asciidoc]: Nit: please wrap at 80 columns.

**Review by jasontedor — COMMENTED** [review]
I left a couple of nits about the column width.

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 10]
```diff
@@ -7,11 +7,27 @@ a time so upgrading does not interrupt service. Running multiple versions of
 not supported, as shards cannot be replicated from upgraded nodes to nodes
 running the older version.
 
-It is best to upgrade the master-eligible nodes in your cluster after all of
-the other nodes. Once you have started to upgrade the master-eligible nodes
-they may form a cluster that nodes of older versions cannot join. If you
-upgrade the master-eligible nodes last then all the other nodes will not be
-running an older version and so they will be able to join the cluster.
+We strongly recommend you upgrade you divide your cluster's nodes into the
```
Comment (jasontedor) [inline docs/reference/upgrade/rolling_upgrade.asciidoc]: "We strongly recommend you upgrade you divide your cluster's nodes" -> "We strongly recommend that when you upgrade you divide your cluster's nodes"

**Review by jasontedor — COMMENTED** [review]
Sorry, one sentence needs a little extra love and care.

**Comment by DaveCTurner** [comment]
Thanks all. @richiejarvis you requested changes but then pushed changes directly to the branch  rather than leaving them as suggestions. I've fixed the formatting and addressed the remaining  comments.

File: docs/reference/upgrade/rolling_upgrade.asciidoc (hunk 10 above)
Comment (DaveCTurner) [reply docs/reference/upgrade/rolling_upgrade.asciidoc]: Good catch, thanks
